service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -170,4 +189,212 @@ void updatePipelineWorkspaceId() {
+    List<PipelineInputDefinition> inputDefinitions = new ArrayList<>(List.of(requiredInput));
+
+    LinkedHashMap<String, Object> inputs =
+        new LinkedHashMap<>(Map.of("required_input", "this is a string"));
+
+    // even though it's the wrong type, the required input is present, so should not return an error
+    // message
+    assertTrue(pipelinesService.validateRequiredInputs(inputDefinitions, inputs).isEmpty());
+  }
+
+  private static Stream<Arguments> inputTypeValidationsValid() {
+    return Stream.of(
+        // arguments are: type specification, value to be tested, isRequired
+        arguments("Integer", 123, true),
+        arguments("Integer", "123", true),
+        arguments("String", "I am a string", true),
+        arguments("VCF", "path/to/file.vcf.gz", true),
+        arguments("Array_String", List.of("this", "is", "a", "list", "of", "strings"), true),
+        arguments(
+            "Array_String",
+            List.of(1, 2, 3),
+            true), // including to remind us that this currently validates
+        arguments("Array_VCF", List.of("path/to/file.vcf.gz"), true),
+        arguments("String", "I am a string", false));
+  }
+
+  @ParameterizedTest
+  @MethodSource("inputTypeValidationsValid")
+  void validateInputTypeValid(String type, Object inputValue, Boolean isRequired) {
+    PipelineInputDefinition requiredInput =
+        new PipelineInputDefinition(1L, "input_name", type, isRequired);
+    List<PipelineInputDefinition> inputDefinitions = new ArrayList<>(List.of(requiredInput));
+
+    LinkedHashMap<String, Object> inputs = new LinkedHashMap<>(Map.of("input_name", inputValue));
+
+    assertTrue(pipelinesService.validateInputTypes(inputDefinitions, inputs).isEmpty());
+  }
+
+  private static Stream<Arguments> inputTypeValidationsNotValid() {
+    String commonTypeErrorMessage = "input_name must be of type %s";
+    String vcfTypeErrorMessage = "input_name must be a path to a VCF file ending in .vcf.gz";
+    String arrayStringTypeErrorMessage = "input_name must be an array of non-null strings";
+    String arrayVCFTypeErrorMessage =
+        "input_name must be an array of non-null paths to VCF files ending in .vcf.gz";
+    String notNullErrorMessage = "input_name must not be null";
+    String notEmptyErrorMessage = "input_name must not be empty";
+    String emptyArrayErrorMessage = "input_name must not be an empty array";
+
+    String[] stringArrayWithNull = new String[] {null, "has a null value"};
+
+    return Stream.of(
+        // arguments are: type specification, value to be tested, isRequired, expected errorMessage
+        // basic type checks
+        arguments("Integer", "I am a string", true, commonTypeErrorMessage),
+        arguments("Integer", 2.3, true, commonTypeErrorMessage),
+        arguments("Integer", "2.3", true, commonTypeErrorMessage),
+        arguments("String", "", true, notEmptyErrorMessage),
+        arguments("String", "", false, notEmptyErrorMessage),
```

Comment:
this is perhaps a misleading test because we're not running the full validation on these, just the validation of type, and the required status is not taken into account in `validateInputTypes`.